indico/modules/events/registration/client/js/components/ParticipantListAccordion.tsx

```diff
@@ -0,0 +1,263 @@
+}
+
+interface ParticipantListAccordionProps {
+  tables: TableObj[];
+}
+
+interface AccordionParticipantsItemProps {
+  index: number;
+  table: TableObj;
+  children?: ReactNode;
+  collapsible?: boolean;
+}
+
+interface ParticipantCounterProps extends HTMLAttributes<HTMLSpanElement> {
+  totalCount: number;
+  hiddenCount: number;
+  styleName?: string;
+}
+
+function ParticipantCountTranslationHidden({count}: {count: number}) {
+  return (
+    <PluralTranslate count={count}>
+      <Singular>
+        <Param name="count" value={count} /> participant registered anonymously.
+      </Singular>
+      <Plural>
+        <Param name="count" value={count} /> participants registered anonymously.
+      </Plural>
+    </PluralTranslate>
+  );
+}
+
+function ParticipantCounter({
+  totalCount,
+  hiddenCount,
+  styleName = '',
+  ...props
+}: ParticipantCounterProps) {
+  const participantCounterElement = (
+    <span>
+      {totalCount} <Icon name="user" />
+    </span>
+  );
+
+  return hiddenCount > 0 ? (
+    <div className={styleName} {...props}>
+      <Popup
+        position="left center"
+        content={<ParticipantCountTranslationHidden count={hiddenCount} />}
+        trigger={participantCounterElement}
+      />
+    </div>
+  ) : (
+    participantCounterElement
+  );
+}
+
+function ParticipantTable({table}: {table: TableObj}) {
+  const visibleParticipantsCount = table.rows.length;
+  const totalParticipantCount = table.num_participants;
+  const hiddenParticipantsCount = totalParticipantCount - visibleParticipantsCount;
+  const hasInvisibleParticipants = hiddenParticipantsCount > 0;
+
+  type sortDirectionType = 'ascending' | 'descending' | null;
+
+  const [sortColumn, setSortColumn] = useState<number | null>(null);
+  const [sortDirection, setSortDirection] = useState<sortDirectionType>(null);
+  const [sortedRows, setSortedRows] = useState([...table.rows]);
+
+  const handleSort = (columnIndex: number) => {
+    const directions: Record<sortDirectionType, sortDirectionType> = {
+      [null as sortDirectionType]: 'ascending',
+      ascending: 'descending',
+      descending: null,
+    };
+
+    const direction: sortDirectionType =
+      sortColumn === columnIndex ? directions[sortDirection] : 'ascending';
+
+    const sortedData =
+      direction === null
+        ? [...table.rows]
+        : [...sortedRows].sort((a, b) => {
+            const textA = a.columns[columnIndex].text.toLowerCase();
+            const textB = b.columns[columnIndex].text.toLowerCase();
+
+            if (textA < textB) {
+              return direction === 'ascending' ? -1 : 1;
+            } else if (textA > textB) {
+              return direction === 'ascending' ? 1 : -1;
+            }
+
+            return 0;
+          });
+
+    setSortColumn(columnIndex);
+    setSortDirection(direction);
+    setSortedRows(sortedData);
+  };
+
+  return visibleParticipantsCount > 0 ? (
+    <Table fixed celled sortable className="table">
```

Comment:
Do we need the `className` here? I think SUI adds the `table` class in this case anyway. Same for `table-row`

Reply:
Seems like I was not even using these classes in my css file so I removed them